### PR TITLE
watermarker allow latency

### PR DESCRIPTION
**Pull request by congdai-airwallex**
we find there is data loss in union operator and we want to allow latency to void late data

### CODE REVIEW

File: core/src/main/java/com/airwallex/airskiff/core/api/Stream.java [hunk 1]
```diff
@@ -95,6 +96,10 @@ default Stream<T> union(Stream<T> that) {
     return new ConcatStream<>(this, that);
   }
 
+  default Stream<T> union(Stream<T> that, Duration allowedLatency) {
```
Comment (lsgrep) [inline core/src/main/java/com/airwallex/airskiff/core/api/Stream.java:99]: It does not feel natural/adequate to put allowedLatency in here. What if we make it a global config for all operations?

Comment (congdai-airwallex) [reply core/src/main/java/com/airwallex/airskiff/core/api/Stream.java:99]: cool, a global config seems more graceful

File: core/src/main/java/com/airwallex/airskiff/flink/HybridWatermarkGenerator.java [hunk 2]
```diff
@@ -12,12 +12,22 @@ public class HybridWatermarkGenerator<T> implements WatermarkGenerator<T> {
   private final Clock clock;
   private long maxTs;
   private long lastProcessTime;
+  private long allowedLatency;
```
Comment (tzhu-airwallex) [inline core/src/main/java/com/airwallex/airskiff/flink/HybridWatermarkGenerator.java:15]: nit: 
```suggestion
  private final long allowedLatency;
```

**Review by tzhu-airwallex — APPROVED** [review]

**Review by tzhu-airwallex — APPROVED** [review]

**Review by lsgrep — APPROVED** [review]
LGTM

**Pull request merged**